src/storage/copier/read_file_state.cpp

```diff
@@ -75,6 +104,67 @@ void ReadParquetSharedState::countNumRows() {
     }
 }
 
+std::unique_ptr<ReadFileMorsel> ReadParquetSharedState::getMorselSerial() {
```

Comment:
You seem to forget to rework this function, as the control logic is different from the csv one. Can you redo it as the csv one?